kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/timing/eventually.kt

```diff
@@ -13,68 +13,69 @@ import kotlin.time.TimeSource
 
 suspend fun <T> eventually(duration: Duration, poll: Duration, f: suspend () -> T): T =
-   eventually(EventuallyConfig(duration = duration, interval = poll.fixed(), exceptionClass = Throwable::class), f = f)
+   eventually(EventuallyConfig(duration, interval = poll.fixed()), f = f)
 
 /**
  * Runs a function until it doesn't throw the specified exception as long as the specified duration hasn't passed
  */
 suspend fun <T> eventually(duration: Duration, exceptionClass: KClass<out Throwable>, f: suspend () -> T): T =
-   eventually(EventuallyConfig(duration = duration, exceptionClass = exceptionClass), f = f)
+   eventually(EventuallyConfig(duration, exceptionClass = exceptionClass), f = f)
 
 /**
  * Runs a function until the following constraints are eventually met:
- * the optional [predicate] must be satisfied, defaults to true
- * the optional [duration] has not passed now, defaults to [Duration.INFINITE]
- * the number of iterations does not exceed the optional [retries], defaults to [Int.MAX_VALUE]
+ * - the optional [predicate] must be satisfied, defaults to true
+ * - the optional [duration] has not passed now, defaults to [Duration.INFINITE]
+ * - the number of iterations does not exceed the optional [retries], defaults to [Int.MAX_VALUE]
  *
- * [eventually] will catch the specified optional [exceptionClass] and (or when not specified) [AssertionError], defaults to [Throwable]
- * [eventually] will delay the specified [interval] between iterations, defaults to 25 [milliseconds]
- * [eventually] will pass the resulting value and state (see [EventuallyState]) into the optional [listener]
+ * eventually will
+ * - catch the specified optional [exceptionClass] and (or when not specified) [AssertionError]
+ * - delay the specified [interval] between iterations, defaults to 25 [milliseconds]
+ *
+ * @return the first accepted result of [f]
  */
 suspend fun <T> eventually(
    duration: Duration = Duration.INFINITE,
    interval: Interval = 25.milliseconds.fixed(),
    predicate: EventuallyPredicate<T> = { true },
-   listener: EventuallyListener<T> = EventuallyListener { },
    retries: Int = Int.MAX_VALUE,
    exceptionClass: KClass<out Throwable>? = null,
    f: suspend () -> T,
-): T = eventually(EventuallyConfig(duration, interval, retries, exceptionClass), predicate, listener, f)
+): T = eventually(EventuallyConfig(duration, interval, retries, exceptionClass), predicate, f)
 
 /**
- * Runs a function until it doesn't throw and the result satisfies the predicate as long as the specified duration hasn't passed
- * and uses [EventuallyConfig] to control the duration, interval, listener, retries, and exceptionClass.
+ * Runs a function until it doesn't throw and the result satisfies the predicate, as long as the specified duration hasn't passed.
+ * @param config controls the duration, interval, retries, and exceptionClass
  */
 suspend fun <T> eventually(
    config: EventuallyConfig,
    predicate: EventuallyPredicate<T> = { true },
```

Comment:
Can we keep the version with a listener, but just deprecate it. It's ok to have copy n paste code to avoid complication, since it will be deleted in a future version.